ubi_manifest/worker/tasks/depsolver/modulemd_depsolver.py

```diff
@@ -0,0 +1,122 @@
+"""
+Module for depsolving modulemds in ubi repositories
+"""
+import logging
+import os
+from itertools import chain
+from typing import Dict, List, Set
+
+from more_executors import Executors
+from more_executors.futures import f_proxy
+from pubtools.pulplib import YumRepository
+
+from .models import ModularDepsolverItem, UbiUnit
+from .pulp_queries import search_modulemds
+from .utils import get_criteria_for_modules, get_modulemd_output_set
+
+_LOG = logging.getLogger(__name__)
+
+MAX_WORKERS = int(os.getenv("UBI_MANIFEST_MODULAR_DEPSOLVER_WORKERS", "8"))
+
+
+class ModularDepsolver:
+    """
+    Class for depsolving modulemd units
+    """
+
+    def __init__(self, modular_items: List[ModularDepsolverItem]) -> None:
+        self._modular_items: List[ModularDepsolverItem] = modular_items
+        self._input_repos: List[YumRepository] = list(
+            chain.from_iterable(item.in_pulp_repos for item in self._modular_items)
+        )
+        # executor for this class, not adding retries because for pulp
+        # we use executor from pulplib
+        self._executor = Executors.thread_pool(max_workers=MAX_WORKERS)
+
+        # set of all already searched modules to avoid duplication & cycles
+        self._searched_modules: Dict[str, Set[str]] = {
+            "without_stream": set(),
+            "with_stream": set(),
+        }
+        # output set of resolved modulemd packages
+        self.modules: List[UbiUnit] = []
+
+    def __enter__(self):
+        return self
```

Comment:
You should probably call `__enter__` on self._executor even if it seems like it does nothing.  Otherwise you call exit on it without ever calling enter, and even if that works on today's version of the code, it might not work in the future.

Reply:
Added self._executor.__enter__()